great_expectations/datasource/fluent/databricks_sql_datasource.py

```diff
@@ -80,14 +82,22 @@ class DatabricksSQLDatasource(SQLDatasource):
     Args:
         name: The name of this DatabricksSQL datasource.
         connection_string: The SQLAlchemy connection string used to connect to the postgres database.
-            For example: "databricks+connector://token:<token>@<host>:<port>/<database>?http_path=<http_path>"
+            For example: "databricks://token:<token>@<host>:<port>/<database>?http_path=<http_path>"
         assets: An optional dictionary whose keys are TableAsset or QueryAsset names and whose values
             are TableAsset or QueryAsset objects.
     """
 
     type: Literal["databricks_sql"] = "databricks_sql"  # type: ignore[assignment]
     connection_string: Union[ConfigStr, DatabricksDsn]
 
+    def test_connection(self, test_assets: bool = True) -> None:
+        try:
+            super().test_connection(test_assets)
+        except TestConnectionError as e:
+            raise TestConnectionError(
+                f"Could not connect to Databricks - please ensure you've installed necessary dependencies with `pip install great_expectations[databricks]`.\n\n{str(e)}"
+            ) from e
```

Comment:
Is there something more specific I can check for so I know when to tell the user to install deps? We wrap everything in `TestConnectionError` so it would cover both dep issues and regular connection issues

Reply:
Example:
```
TestConnectionError: Could not connect to Databricks - please ensure you've installed necessary dependencies with `pip install great_expectations[databricks]`.

Attempt to connect to datasource failed with the following error message: Unable to create a SQLAlchemy engine from connection_string: databricks+connector://token:dapi227022d42e940abd52aad081e0079208@dbc-8ad5a215-e8c7.cloud.databricks.com:443/cloud_events?http_path=/sql/1.0/warehouses/0f7900ef66eb08c3 due to the following exception: Can't load plugin: sqlalchemy.dialects:databricks.connector
```

Reply:
You should be able to programmatically access the underlying exception.
Need to look that up....

Reply:
```python
def nested_exception():
    try:
        "1" + 2
    except TypeError as t_err:
        raise ValueError("Oops") from t_err

try:
    nested_exception()
except Exception as exc:
    print(repr(exc))
    print(repr(exc.__cause__))
    print(exc.__cause__.__cause__)

... ValueError('Oops')
... TypeError('can only concatenate str (not "int") to str')
... None
```